src/app.js

```diff
@@ -335,6 +276,192 @@ fluid.onUncaughtException.addListener(function (err) {
+        timeout: 0
+    },
+    members: {
+        dialog: {
+            expander: {
+                funcName: "gpii.app.makeWaitDialog"
+            }
+        }
+    },
+    modelListeners: {
+        "showDialog": {
+            funcName: "gpii.app.showHideWaitDialog",
+            args: ["{that}", "{change}.value"]
+        }
+    },
+    listeners: {
+        "onDestroy": {
+            funcName: "clearTimeout",
+            args: ["{that.model.timeout}"]
+        }
+    }
+});
+
+/**
+ * Creates a dialog. This is done up front to avoid the delay from creating a new
+ * dialog every time a new message should be displayed.
+ */
+gpii.app.makeWaitDialog = function () {
+    var screenSize = electron.screen.getPrimaryDisplay().workAreaSize;
+
+    var dialog = new BrowserWindow({
+        show: false,
+        frame: false,
+        transparent: true,
+        alwaysOnTop: true,
+        skipTaskBar: true,
+        x: screenSize.width - 900, // because the default width is 800
+        y: screenSize.height - 600 // because the default height is 600
+    });
+
+    var url = fluid.stringTemplate("file://%dirName/html/message.html", {
+        dirName: __dirname
+    });
+    dialog.loadURL(url);
+    return dialog;
+};
+
+
+gpii.app.showHideWaitDialog = function (that, showDialog) {
+    showDialog ? gpii.app.displayWaitDialog(that) : gpii.app.dismissWaitDialog(that);
+};
+
+/**
+ * Shows the dialog on users screen with the message passed as parameter.
+ * Records the time it was shown in `dialogStartTime` which we need when
+ * dismissing it (checking whether it's been displayed for the minimum amount of time)
+ *
+ * @param that {Object} the app module
+ */
+gpii.app.displayWaitDialog = function (that) {
+    that.dialog.show();
+    // Hack to ensure it stays on top, even as the GPII autoconfiguration starts applications, etc., that might
+    // otherwise want to be on top
+    // see amongst other: https://blogs.msdn.microsoft.com/oldnewthing/20110310-00/?p=11253/
+    // and https://github.com/electron/electron/issues/2097
+    var interval = setInterval(function () {
+        if (!that.dialog.isVisible()) {
+            clearInterval(interval);
+        };
+        that.dialog.setAlwaysOnTop(true);
+    }, 100);
+
+    that.model.dialogStartTime = Date.now();
+};
+
+/**
+ * Dismisses the dialog. If less than `that.dialogMinDisplayTime` ms have passed since we first displayed
+ * the window, the function waits until `dialogMinDisplayTime` has passed before dismissing it.
+ *
+ * @param that {Object} the app
+ */
+gpii.app.dismissWaitDialog = function (that) {
+    // ensure we have displayed for a minimum amount of `dialogMinDisplayTime` secs to avoid confusing flickering
+    var remainingDisplayTime = (that.model.dialogStartTime + that.model.dialogMinDisplayTime) - Date.now();
+
+    if (remainingDisplayTime > 0) {
+        that.model.timeout = setTimeout(function () {
```

Comment:
One musn't write directly into a model by assignment - this kind of free-form material belongs at top level on the component or as a "member". Also, please give this a more descriptive name than "timeout". In fact the leak causing the failure isn't at this site but above it on line 436, but it is good to clean up this leak as well.